packages/editor/src/components/post-taxonomies/style.scss

```diff
@@ -1,3 +1,8 @@
+.editor-post-taxonomies__hierarchical-terms-list {
+	max-height: 14em;
+	overflow: auto;
```

Comment:
scrollable divs need to be made keyboard accessible: only Firefox makes them focusable by default so for other browsers it's necessary to make them focusable and add some ARIA. See for example the Inserter. I'd say something like `tabIndex="0" role="group" aria-label={ __( 'Available terms' ) }"` could work